checks-superstaq/checks_superstaq/license_header_format_.py

```diff
@@ -0,0 +1,546 @@
+        )
+
+
+def _extract_license_header(file: str) -> list[LicenseHeader]:
+    """Extracts the license headers from a file. Reads the file until it finds a none comment line.
+    Pylint and mypy disabling comments and shebangs are ignored. White spaces preciding the
+    license header and between the header and shebang and/or pylint line are also ignored.
+    Also checks if the comment block selected contains the keyword 'Copyright'.
+
+    Args:
+        file: The file name/path.
+
+    Returns a list of LicenseHeader object each for being the distinct license headers found in
+    the file.
+    """
+    license_header_lst: list[LicenseHeader] = []
+    license_header = ""
+    exceptions = ["# pylint:", "#!/", "# mypy:"]
+
+    with open(file, "r") as f:
+        for line_num, line in enumerate(f):
+            if not license_header and line[0] == "\n":
+                continue
+            if line[0] != "#" and line[0] != "\n":
+                if license_header:
+                    license_header_lst[-1].license_header = license_header
+                    license_header_lst[-1].end_line_num = line_num + 1
+                    license_header = ""
+                break
+
+            if all(exception not in line for exception in exceptions):
+                if not license_header:
+                    license_header_lst.append(LicenseHeader(line_num + 1))
+
+                if line == "\n":
+                    # set the line number for the last line of the license_header
+                    license_header_lst[-1].license_header = license_header.strip()
+                    license_header_lst[-1].end_line_num = line_num + 1
+                    license_header = ""
+                else:
+                    license_header += line
+    license_header_lst = [
+        header for header in license_header_lst if "Copyright" in header.license_header
+    ]
+
+    return license_header_lst
+
+
+def _validate_license_header(
+    license_header_lst: list[LicenseHeader],
+    expected_license_header: str,
+    licensee: str,
+    editable: bool,
+) -> bool:
+    """Returns whether there is a valid license header in a file and for each license
+    header in a file, it assigns each theiir type.
+        - VALID: if the header contains a Copyright line.
+        - OUTDATED: if the header has the correct licensee but different type.
+        - SIMILAR_LICENSE: if the header is a valid license but not from the licensee.
+        - OTHER: if the header is any other one.
+    Args:
+        license_header_lst: List of the license_header objects for each header in a file.
+        licensee: The owner of the expected license.
+        editable: Whether similar licenses can be edited to include the license owner instead
+            adding the entire license to the file.
+
+    Returns: Whether there is a valid license header in a file or not.
+    """
+    copyright_line = ""
+    body = ""
+    copyright_pattern = re.compile(r"Copyright .*")
```

Comment:
should we make this one case insensitive? e.g.
```suggestion
    copyright_pattern = re.compile(r"Copyright .*", flags=re.IGNORECASE)
```